### DISCUSSION TITLE
schema for all things defined as default

### DISCUSSION THREAD

**Discussion by listaction** [data-driven-forms/react-forms#1013] [Need help] [answered]
Hi Folks, 
 I was looking for a project that did exactly this - looks very promising - is there a schema for all things defined in data-driver-forms that's available as a URL ? I realize it can be extended - but I am looking for some kitchen sink schema that defines the various properties and types so clearly explained in the getting started under a single link.

Cheers,
K

**Comment by rvsia** [comment, marked as answer]
Hello,

I think the closest list to the list of generic attributes is this one: https://data-driven-forms.org/schema/introduction

**Reply by listaction** [reply]
thanks @rvsia ! 🙇‍♂️

**Reply by malloy045** [reply]
@rvsia, I assume this means there is no formal [JSON Schema](https://json-schema.org/) (that represents all possible options for a field schema) that someone could use for validation and code generation?

**Reply by rvsia** [reply]
Yes, you are correct. Unfortunately, we do not have schema like that. However, we provide TypeScript types that can help a little bit.

**Reply by malloy045** [reply]
> Yes, you are correct. Unfortunately, we do not have schema like that. However, we provide TypeScript types that can help a little bit

Thanks. Is there any way you could drop a link to the TypeScript models in the repo or documentation?

**Reply by rvsia** [reply]
Each file in the repository should have associated `d.ts` files. For example https://github.com/data-driven-forms/react-forms/blob/master/packages/react-form-renderer/src/form-renderer/form-renderer.d.ts

TypeScripts loads these files automatically.
